naas/requests/campaigns.py

```diff
@@ -0,0 +1,94 @@
+from naas.client import Client
+
+
+class Campaigns:
+
+    @staticmethod
+    def list_by_project_id(project_id, params=None):
+        """
+        Retrieve the list of campaigns by project
+        :param project_id: string
+        :param params: dict
+        :return: Response
+        """
+        if params is None:
+            params = {}
+        rel = Client.rel_for('rels/project-campaigns')
+        route = Client.routes().route_for(rel)
+        url = route.url_for(args=params.update({'project_id': project_id}))
```

Comment:
@muhallan - I noticed an issue here. `update` returns `None` when run in place like this, so it never adds the `project_id` to the route. If I separated this out it works:

```py
route_params = params.update({ 'project_id': project_id })
url          = route.url_for(route_params)
```

This is across the board for all **Request** objects.

Reply:
Yeah, just realized this. I am going to fix all of them properly. Thanks.